### PR TITLE
Add list-virtual-routers and CRUD redistribution profile commands - pan-os

### PR DESCRIPTION
**Pull request by GuyAfik**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
relates: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-914)

## Description
Add CRUD (create, delete, update, list) operations for redistribution profile and listing all the virtual-routers in pan-os

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 1]
```diff
@@ -11191,19 +11193,6 @@ def parse_pan_os_un_committed_data(dictionary, keys_to_remove):
                     parse_pan_os_un_committed_data(item, keys_to_remove)
 
 
-def extract_objects_info_by_key(_entry, _key):
```
Comment (JudahSchwartz) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.py:11194]: looks like youre deleting this but I think you added it in the last pr

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 2]
```diff
@@ -11317,10 +11297,6 @@ def pan_os_list_nat_rules_command(args):
     )
 
 
-def dict_to_xml(_dictionary):
```
Comment (JudahSchwartz) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.py:11320]: this too

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 3]
```diff
@@ -11622,6 +11556,451 @@ def pan_os_edit_nat_rule_command(args):
...
+        return xml.replace('&gt;', '>').replace('&lt;', '<')
+    return xml
+
+
+def pan_os_create_redistribution_profile(args):
+    def _set_up_body_request():
+        def _set_up_ospf_filter_body_request():
+            _ospf_filter_body_request = {}
+            if filter_ospf_path_type := args.get('filter_ospf_path_type'):
+                _ospf_filter_body_request.update(prepare_pan_os_objects_body_request('path-type', filter_ospf_path_type))
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/PAN-OS/Integrations/Panorama/Panorama.py:11796]: `prepare_pan_os_objects_body_request` has the variable `is_list=True` by default. so its like passing in `is_list=True` when not stating it directly.

some of the pan-os objects can be lists while some cannot. This is something that we know from how the api works + viewing at the user-interface. if its confusing that by default it gets `True`, i can add to specifically add `is_list=True` to every call, although its redundant.

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 4]
```diff
@@ -11622,6 +11556,451 @@ def pan_os_edit_nat_rule_command(args):
...
+        raise ValueError(f'behavior argument must be one of replace/remove/add values')
+
+    if behavior == 'replace':
+        element = prepare_pan_os_objects_body_request(object_name, element_value, is_list=is_listable)
+    else:  # add or remove is only for listable objects.
+        current_objects_items = panorama_get_current_element(element_to_change=object_name, xpath=xpath)
+        if behavior == 'add':
+            updated_object_items = list((set(current_objects_items)).union(set(argToList(element_value))))
+        else:  # remove
+            updated_object_items = [item for item in current_objects_items if item not in element_value]
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/PAN-OS/Integrations/Panorama/Panorama.py:11932]: you are right, should be `argToList`, nice catch.

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (JudahSchwartz) [reply Packs/PAN-OS/Integrations/Panorama/Panorama.py:11796]: ignore this . I misread the function

**Review by JudahSchwartz — APPROVED** [review]

**Pull request merged**